src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -132,16 +132,16 @@ def read_file_content(file_path, allow_binary=False):
             with codecs_open(file_path, encoding=encoding) as f:
                 logger.debug("attempting to read file %s as %s", file_path, encoding)
                 return f.read()
-        except UnicodeDecodeError:
-            if allow_binary:
-                with open(file_path, 'rb') as input_file:
-                    logger.debug("attempting to read file %s as binary", file_path)
-                    return base64.b64encode(input_file.read()).decode("utf-8")
-            else:
-                raise
-        except UnicodeError:
+        except (UnicodeError, UnicodeDecodeError):
             pass
 
+    if allow_binary:
+        try:
+            with open(file_path, 'rb') as input_file:
+                logger.debug("attempting to read file %s as binary", file_path)
+                return base64.b64encode(input_file.read()).decode("utf-8")
+        except:  # pylint: disable=bare-except
```

Comment:
You need to at least catch  `Exception`; otherwise the code will capture keyboard interrupt